### ISSUE TITLE
Support docblock inheritance

### ISSUE THREAD

**Issue by GiacoCorsiglia** [bmewburn/intelephense#36]
Hi there,

This is a fantastic extension, it "just works".  I have a feature request: to add support for docblock inheritance, as defined [here](https://www.phpdoc.org/docs/latest/guides/inheritance.html).  For example:

```php
$child = new Child;
// Should inherit documentation for this method from Parent::method
$child->method();

class Parent {
  /**
   * Parent method documentation.
   */
  public function method() {}
}

class Child extends Parent {
  public function method() {/* overridden behavior */}
}
```

A stretch goal would be actually respecting the `{@inheritDoc}` tag, so child classes can override just part of the documentation.

I would be happy to take a look at this myself if you could point me in the right direction.

Thanks!

**Comment by bmewburn** [comment]
This would be a nice feature to have. I appreciate you looking into it and hope the following notes help. Feel free to ask any questions.  

`PhpSymbol` in `symbol.ts` is the interface that holds all symbol information and the symbol tree for a file is built with the `SymbolReader` class in `symbolReader.ts`. I'm refactoring `PhpSymbol` a little right now so that it has a `PhpSymbol.doc` property which will hold docblock info. Currently docblock info is stored in `PhpSymbol.description` and `PhpSymbol.type`. I think it will be easier to work with the revised interface, but it can be easily changed later if you want to start before I get the refactor done.

During the symbol tree construction a method docblock, for example, would be parsed by `PhpDocParser` and have the summary, description and `@return` type assigned to the method symbol, whilst child parameter symbols would have the `@param` type and description assigned to them. I'm unfamiliar with where the `{@inheritDoc}` tag is allowed within a docblock. You may want to check the docblock parser to see if it can handle it as it is quite basic. If it is only expected within the main description and/or tag description sections then it's probably fine, otherwise the current regexes that pull out tag info may not work.

For the document inheritance, an approach that comes to mind would be to create a merge symbols function that takes an array of symbols found in the inheritance chain and produces a unique array of symbols with the merged information. Symbols should be treated as read only, so create new symbols when merging. The order of the passed in symbols can be guaranteed to follow the inheritance order to make processing easier. The passed in array could conceivably contain all methods in the inheritance chain not just all methods of a particular name. 

The merge symbols method could then be used in the various providers or perhaps be centralised in `SymbolStore`. See `SymbolStore.lookupTypeMember` in `symbolStore.ts` which is a method used by many of the providers to get member info. #26 may tie into this feature too, though I'm unsure it it's isolated to the completion provider.

The code has a few rough edges and is in constant development. I hope you can find your way around ok. I'm open to any suggestion you have. Thanks!

**Comment by bmewburn** [comment]
@GiacoCorsiglia just a note to let you know doc block inheritance and `{@inheritdoc}` has been resolved in upcoming v0.8.0

**Comment by bmewburn** [comment]
resolved in v0.8.0
